### DISCUSSION TITLE
determine the number of unique image colors

### DISCUSSION THREAD

**Discussion by samssonn** [dlemstra/Magick.NET#592] [General]
How to determine the number of unique image colors? I tried the image.UniqueColors () method; but it returns the number of all colors in the image. And you need to get the number of unique colors.

**Comment by dlemstra** [comment]
It seems that this question continued here: https://imagemagick.org/discourse-server/viewtopic.php?f=27&t=37711

**Comment by samssonn** [comment]
Yes, but I never received a normal answer.

**Comment by dlemstra** [comment]
What do you mean by a normal answer? What kind of information are you missing in the provided answers?

**Comment by samssonn** [comment]
There is a color image in which there are millions of different colors and shades. I need to get the number of unique colors, that is, those that do not repeat. For example, with the existing million colors and shades, there are repetitions, and there are also those that are not repeated, these are unique colors. How do I get the number of these colors in an image? I need C # code.

**Comment by dlemstra** [comment]
The `UniqueColors` method return the unqiue colors. But due to your input image that might be all the colors of the image. Even a tiny different in pixel colors will cause it to detect difference. I don't understand what you are expecting?

**Comment by samssonn** [comment]
Could you please example code C # How to use the method? It is necessary that the variable has a number of unique colors.

**Comment by dlemstra** [comment]
The method `UniqueColors` returns the number of unique colors. And we tried to explain why you think that the result is wrong. But the result of that method is what you are looking for.

**Comment by samssonn** [comment]
Could you please sample code C #
How to use it?

**Comment by dlemstra** [comment]
The `UniqueColors()` method will return an image with the unique number of colors. To get the total you will need to do something like this:

```C#
string InputImagePath = @"D:\User\1.jpg";
using (MagickImage image = new MagickImage(InputImagePath))
{
    using (var uniqueColors = image.UniqueColors())
    {
        return uniqueColors.Width; // Each color is one pixel.
    }
}
```

**Reply by FurkanGozukara** [reply]
I think image.TotalColors gives unique colors. I tested getting same value always

**Comment by samssonn** [comment]
Your code returns the width of the image. And I need a number of unique colors.

**Comment by dlemstra** [comment]
I made a typo it should be `uniqueColors.Width` instead.

**Comment by samssonn** [comment]
Thank you so much, you helped me a lot.
